coremltools/converters/nnssa/coreml/graph_pass/mlmodel_passes.py

```diff
@@ -289,10 +289,13 @@ def _find_redundant_transposes(nn_spec):
             else:
                 # Check if the layers are connected in the graph.
                 connected = (previous_transpose['layer'].output == _layer.input)
+
                 # Check if they're each other's inverses.
-                this_transpose = _layer.transpose.axes
-                composed = [previous_transpose['axes'][i] for i in this_transpose]
-                compose_to_identity = all([ax == i for i, ax in enumerate(composed)])
+                if connected:
```

Comment:
@sberardi-apple  I think the compose_to_identy should only check if there are connect.
I got some out of index error on line 296.

Reply:
yes, I think you're right